src/Language/Fixpoint/Solver/Solution.hs

```diff
@@ -296,19 +300,32 @@ apply g s bs      =
   -- Clear the "known" bindings for applyKVars, since it depends on
   -- using the fully expanded representation of the predicates to bind their
   -- variables with quantifiers.
-  do (ps,  ks) <- envConcKVars g s bs
+  do xrs <- traverse (lookupBindEnvExt g s) (F.elemsIBindEnv bs)
+     let (ps,  ks) = envConcKVars xrs
      (pks, kI) <- applyKVars g {ceBindingsInSmt = F.emptyIBindEnv} s ks
      pure (F.conj (pks:ps), kI)   -- see [NOTE: pAnd-SLOW]
 
+-- | @applyInSortedReft@ applies the solution to a single sorted reft
+--
+-- At the time of writing this function is used in PLE, where we need the
+-- expression in unelaborated form. Thus the result is not elaborated here.
+--
+applyInSortedReft
+  :: CombinedEnv ann
+  -> Sol.Sol Sol.QBind
+  -> (F.Symbol, F.SortedReft)
+  -> ElabM (F.Symbol, F.SortedReft)
+applyInSortedReft g s xsr@(x, sr) =
+  do let (ps,  ks) = envConcKVars [xsr]
+     (pks, _) <- applyKVars g {ceBindingsInSmt = F.emptyIBindEnv} s ks
+     pure (x, sr { F.sr_reft = F.Reft (x, F.conj (So.unElab pks : ps)) })
```

Comment:
I'm calling `unElab`. Maybe I should try instead to stop `applyKVars` from elaborating the term. (?)